### ISSUE TITLE
Bug: table has double prefix when LIKE clause

### ISSUE THREAD

**Issue by iRedds** [codeigniter4/CodeIgniter4#5775]
### PHP Version

7.3

### CodeIgniter4 Version

4.1.9 + dev

### CodeIgniter4 Installation Method

Git

### Which operating systems have you tested for this bug?

Windows

### Which server did you use?

cli

### What happened?

see https://forum.codeigniter.com/thread-81454.html

### Steps to Reproduce

```php
// database.default.DBPrefix = db_

db_connect()->table('test')->like('test.field', 'string')->getCompiledSelect();

// SELECT * FROM `db_test` WHERE `db_``db_test`.`field` LIKE '%string%' ESCAPE '!' 

```

### Expected Output

```
SELECT * FROM `db_test` WHERE `db_test`.`field` LIKE '%string%' ESCAPE '!' 
```

**Comment by joaisbn** [comment]
I had the same error, using CI 4.1.9.

I think it can be solved replacing the operator !== ( not identical ) to != ( not equal ) in the BaseConnection.php on line 1061.

`elseif (strpos($parts[$i], $this->DBPrefix) !== 0) {`
to
`elseif (strpos($parts[$i], $this->DBPrefix) != 0) {`

**Comment by kenjis** [comment]
@joaisbn This was fixed by  #5778
Try the latest `develop` branch.
